test/functional/page_objects/visual_builder_page.ts

```diff
@@ -453,6 +453,14 @@ export function VisualBuilderPageProvider({ getService, getPageObjects }: FtrPro
       await PageObjects.header.waitUntilLoadingHasFinished();
     }
 
+    public async waitForIndexPatternTimeFieldOptionsLoaded() {
+      await retry.waitFor('combobox options loaded', async () => {
+        const options = await comboBox.getOptions('metricsIndexPatternFieldsSelect');
+        log.debug(`-- optionsCount=${options.length}`);
+        return options.length > 0;
+      });
+    }
```

Comment:
Do you think we need a retry here? I wonder if there is a chance that option list will be updated without field value being re-typed?

Looks like `openOptionsList` is not reopening list, if it is already displayed. 

It's not critical if it works stable, just want to point out that using nested retries might slow things down.

Reply:
Yes I think it is as `await comboBox.getOptions('metricsIndexPatternFieldsSelect');` can return 0. The problem is that the `metricsIndexPatternFieldsSelect` dropdown has a delay on filling in with the options so here the retry helps to get the options and when there are > 0 it will continue to the rest of the test